src/rpc/daemon_handler.cpp

```diff
@@ -0,0 +1,726 @@
+  {
+    res.height = m_core.get_current_blockchain_height();
+
+    res.target_height = m_core.get_target_blockchain_height();
+
+    auto& chain = m_core.get_blockchain_storage();
+
+    res.difficulty = chain.get_difficulty_for_next_block();
+
+    res.target = chain.get_difficulty_target();
+
+    res.tx_count = chain.get_total_transactions() - res.height; //without coinbase
+
+    res.tx_pool_size = m_core.get_pool_transactions_count();
+
+    res.alt_blocks_count = chain.get_alternative_blocks_count();
+
+    uint64_t total_conn = m_p2p.get_connections_count();
+    res.outgoing_connections_count = m_p2p.get_outgoing_connections_count();
+    res.incoming_connections_count = total_conn - res.outgoing_connections_count;
+
+    res.white_peerlist_size = m_p2p.get_peerlist_manager().get_white_peers_count();
+
+    res.grey_peerlist_size = m_p2p.get_peerlist_manager().get_gray_peers_count();
+
+    res.testnet = m_core.is_testnet();
+  }
+
+  void DaemonHandler::handle(StopMining::Request& req, StopMining::Response& res)
+  {
+  }
+
+  void DaemonHandler::handle(MiningStatus::Request& req, MiningStatus::Response& res)
+  {
+  }
+
+  void DaemonHandler::handle(SaveBC::Request& req, SaveBC::Response& res)
+  {
+    if (!m_core.get_blockchain_storage().store_blockchain())
+    {
+      res.status = Message::STATUS_FAILED;
+      res.error_details = "Error storing the blockchain";
+    }
+    else
+    {
+      res.status = Message::STATUS_OK;
+    }
+  }
+
+  void DaemonHandler::handle(GetBlockHash::Request& req, GetBlockHash::Response& res)
+  {
+    if (m_core.get_current_blockchain_height() <= req.height)
+    {
+      res.hash = cryptonote::null_hash;
+      res.status = Message::STATUS_FAILED;
+      res.error_details = "height given is higher than current chain height";
+      return;
+    }
+
+    res.hash = m_core.get_block_id_by_height(req.height);
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(GetBlockTemplate::Request& req, GetBlockTemplate::Response& res)
+  {
+  }
+
+  void DaemonHandler::handle(SubmitBlock::Request& req, SubmitBlock::Response& res)
+  {
+  }
+
+  void DaemonHandler::handle(GetLastBlockHeader::Request& req, GetLastBlockHeader::Response& res)
+  {
+    res.hash = m_core.get_tail_id();
+
+    GetBlockHeaderByHash::Request fake_req;
+    fake_req.hash = res.hash;
+
+    GetBlockHeaderByHash::Response fake_res;
+
+    handle(fake_req, fake_res);
+
+    if (!fake_res.error_details.empty())
+    {
+      res.status = fake_res.status;
+      res.error_details = fake_res.error_details;
+      return;
+    }
+
+    res.major_version = fake_res.major_version;
+    res.minor_version = fake_res.minor_version;
+    res.timestamp = fake_res.timestamp;
+    res.prev_id = fake_res.prev_id;
+    res.nonce = fake_res.nonce;
+    res.height = fake_res.height;
+    res.difficulty = fake_res.difficulty;
+    res.reward = fake_res.reward;
```

Comment:
Good point.  Took me a bit to figure out how I wanted to handle that.  I think I've sorted it in a reasonable way, let me know.